python/lsst/daf/butler/core/dimensions/_records.py

```diff
@@ -159,6 +159,38 @@ class Config:
             }
         }
 
+    @classmethod
+    def direct(cls, *, definition, record) -> SerializedDimensionRecord:
+        """Construct a `SerializedDimensionRecord` directly without validators.
+
+        This differs from PyDantics construct method in that the arguments are
+        explicitly what the model requires, and it will recurse through
+        members, constructing them from their corresponding `direct` methods.
+
+        This method should only be called when the inputs are trusted.
+        """
+        node = SerializedDimensionRecord.__new__(cls)
+        setter = object.__setattr__
+        setter(node, 'definition', definition)
+        setter(node, 'record', record)
+        setter(node, '__fields_set__', {'definition', 'record'})
+        return node
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, SerializedDimensionRecord):
+            return NotImplemented
+        if self.definition != other.definition:
+            return False
+        if self.record.keys() != other.record.keys():
+            return False
+        for key in self.record:
+            if self.record[key] != other.record[key]:
+                return False
+        return True
+
+    def __hash__(self) -> int:
+        return hash((self.definition, tuple((k, v) for k, v in self.record.items())))
```

Comment:
I'd recommend removing this (and `__eq__`) after you switch to the autoincrement keys in serialization.